compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/WebSocketUpgradeServiceValidator.java

```diff
@@ -94,8 +94,11 @@ void validate() {
             for (ReturnStatementNode returnStatementNode : returnStatementNodeVisitor.getReturnStatementNodes()) {
                 ExpressionNode expressionNode = returnStatementNode.expression().get();
                 if (expressionNode instanceof NewExpressionNode) {
-                    final TypeReferenceTypeSymbol definition = (TypeReferenceTypeSymbol) ctx.semanticModel()
-                            .type(expressionNode).get();
+                    TypeSymbol typeSymbol = ctx.semanticModel().type(expressionNode).get();
+                    if (typeSymbol.typeKind() == TypeDescKind.UNION) {
+                        return;
+                    }
+                    final TypeReferenceTypeSymbol definition = (TypeReferenceTypeSymbol) typeSymbol;
```

Comment:
Can't there be other type rather than `TypeReferenceTypeSymbol` ? If so, better if we could add some kind of a validation here.

Reply:
Addressed the suggestions with [8b4d133](https://github.com/ballerina-platform/module-ballerina-websocket/pull/551/commits/8b4d1331265fd97a916a6e71c4786824fca6e88f)